src/Geocoder/Provider/Mapzen.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/**
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+namespace Geocoder\Provider;
+
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\QuotaExceeded;
+use Geocoder\Exception\NoResult;
+use Geocoder\Exception\UnsupportedOperation;
+use Http\Client\HttpClient;
+
+/**
+ * @author Gary Gale <gary@vicchi.org>
+ */
+class Mapzen extends AbstractHttpProvider
```

Comment:
> I would have thought it would make more sense to apply finalisation across the board, rather than have a single provider finalise and none of the others doing so

definitely yes!